src/composer.ts

```diff
@@ -773,10 +773,7 @@ export class Composer<C extends Context> implements Middleware.Obj<C> {
           throw new Error('next() called multiple times')
         }
         index = i
-        const handler = middlewares[i] ? Composer.unwrap(middlewares[i]) : next
-        if (!handler) {
-          return
-        }
+        const handler = Composer.unwrap(middlewares[i] ?? next)
```

Comment:
Why would `next` need to be passed to `Composer.unwrap`? It won't be `Middleware.Obj`, will it?

Reply:
It won't be `Middleware.Obj`, but it actually happens to be assignable to `Middleware.Fn`.

It's passed to `Composer.unwrap` because `middlewares[i]` (now typed as possibly `undefined`) is.

Reply:
So what's wrong with
```suggestion
        const handler = middlewares[i] === undefined ? next : Composer.unwrap(middlewares[i])
```
?

Reply:
TypeScript is not smart enough to be able to narrow one element of array, I'd need to introduce a `const` for `middleares[i]`

Reply:
I find two lines of code still better than doing that function call for `next`. That's confusing because it's not necessary.

I know that this is a nitpick ;)

Reply:
I found a better way to write this:

```ts
        const handler = middlewares[i]
        if (handler === undefined) return next()
        await Composer.unwrap(handler)(context, async (ctx = context) => {
          await execute(i + 1, ctx)
        })
```